projects/adapter/src/dbt/adapters/fal_experimental/adapter.py

```diff
@@ -80,136 +76,65 @@ def run_in_environment_with_adapter(
 
     The environment_name must be defined inside fal_project.yml file
     in your project's root directory."""
-    if type(environment) in [IsolateServer, KoldstartDummyServer]:
-        if type(config.credentials).__name__ == 'BigQueryCredentials' and str(config.credentials.method) == 'service-account':
-            raise RuntimeError(
-                "BigQuery credential method `service-account` is not supported." + \
-                " Please use `service-account-json` instead")
-        deps = [
-            i
-            for i in get_default_pip_dependencies(is_remote=True, adapter_type=adapter_type)
-        ]
-
-        extra_config = {
-            'kind': 'virtualenv',
-            'configuration': { 'requirements': deps }
-        }
+    compressed_local_packages = None
 
-        environment.target_environments.append(extra_config)
+    deps = get_default_pip_dependencies(
+        is_remote=(type(environment.host)==KoldstartHost),
```

Comment:
NIT: you are doing `type ... is` for the same check a few lines below. Maybe store an `is_remote` variable and use it everywhere?